frontend/src/App.js

```diff
@@ -0,0 +1,457 @@
+      return productlist.sort((a, b) => b.price - a.price);
+    }
+    if (sortBy === "PRICE_LOW_TO_HIGH") {
+      return [...productlist].sort((a, b) => a.price - b.price);
+    }
+    return productlist;
+  }
+
+  function getfilteredData(
+    productlist,
+    {
+      showFastDeliveryOnly,
+      showInventoryAll,
+      showMaterial,
+      showIdealFor,
+      showDiscount,
+      showNew
+    }
+  ) {
+    // console.log(">", showDiscount);
+    return productlist
+      .filter((item) => (showInventoryAll ? item.inStock : true))
+      .filter((item) => (showFastDeliveryOnly ? item.fastDelivery : true))
+      .filter((item) =>
+        showIdealFor === "All" ? item.idealFor : item.idealFor === showIdealFor
+      )
+      .filter((item) =>
+        showMaterial === "All" ? item.material : item.material === showMaterial
+      )
+      .filter((item) =>
+        showDiscount === 0 ? item.offer : item.offer === showDiscount
+      )
+      .filter((item) => (showNew === false ? true : item.isnew === showNew));
+  }
+
+  //  console.log("state.sortby", sortBy);
+
+  let sortedData = getSortedData(data, sortBy);
+  let filteredData = getfilteredData(sortedData, {
+    showFastDeliveryOnly,
+    showInventoryAll,
+    showMaterial,
+    showIdealFor,
+    showDiscount,
+    showNew
+  });
+  // console.log("lll", filteredData);
+  // console.log("l", showDiscount);
+  async function cart_add_call(url, payload) {
+    try {
+      let response = await axios.post(url, payload);
+      if (response.status === 200) {
+        dispatch({
+          type: "ADD_TO_CART",
+          payload: { itemId: payload.items.itemId, qty: payload.items.qty }
+        });
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  }
+  async function cart_increase_call(url, payload) {
+    try {
+      let response = await axios.put(url, payload);
+      if (response.status === 200) {
+        dispatch({
+          type: "INCREASE_IN_CART",
+          payload: payload.items.itemId
+        });
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  async function cart_decrease_call(url, payload) {
+    try {
+      let response = await axios.put(url, payload);
+      if (response.status === 200) {
+        dispatch({
+          type: "DECREASE_IN_CART",
+          payload: payload.items.itemId
+        });
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  }
+  async function cart_remove_call(url, payload) {
+    console.log("payload", payload);
+    try {
+      let response = await axios.delete(url, { data: payload });
+      if (response.status === 200) {
+        dispatch({
+          type: "REMOVE_FROM_CARTLIST",
+          payload: payload.itemId
+        });
+      }
+    } catch (err) {
+      console.log("err", err);
+    }
+  }
+  function Add_to_cart_button(
+    id,
+    name,
+    image,
+    price,
+    productName,
+    inStock,
+    level,
+    fastDelivery,
+    isnew,
+    ratings,
+    offer
+  ) {
```

Comment:
Some of the arguments are unused here so you can remove them.

Reply:
I will remove unused arguments.